lib/core/services/alfresco-api.service.ts

```diff
@@ -79,15 +109,21 @@ export class AlfrescoApiService {
             domainPrefix : this.appConfig.get<string>(AppConfigValues.STORAGE_PREFIX),
             oauth2: oauth
         });
+    }
 
-        if (this.alfrescoApi && this.isDifferentConfig(this.lastConfig, config)) {
-            this.lastConfig = config;
-            this.alfrescoApi.setConfig(config);
+    protected initAlfrescoApi() {
+        this.getCurrentAppConfig();
+        this.initAlfrescoApiWithLastConfig();
+    }
+
+    private initAlfrescoApiWithLastConfig() {
+        if (this.alfrescoApi && this.isDifferentConfig(this.lastConfig, this.currentAppConfig)) {
+            this.lastConfig = this.currentAppConfig;
+            this.alfrescoApi.setConfig(this.currentAppConfig);
         } else {
-            this.lastConfig = config;
-            this.alfrescoApi = new AlfrescoApi(config);
+            this.lastConfig = this.currentAppConfig;
```

Comment:
you can put this code before the if else statement so you write it only once

Reply:
I cannot put it before the if otherwise, they will never be different. Let me check why we